main.py

```diff
@@ -48,177 +62,352 @@ def validate_yaml_file(yaml_path: Path) -> dict:
 
     @staticmethod
     def validate_config(config_yaml_path: Path) -> dict:
+        """
+        Validates the configuration file to ensure all required keys are present
+        and have the correct types.
+        """
+        logger.info(f"Loading and validating config file: {config_yaml_path}")
         parameters = ConfigValidator.validate_yaml_file(config_yaml_path)
+        logger.debug(f"Loaded config file content: {parameters}")
+
         required_keys = {
-            'remote': bool,
-            'experience_level': dict,
-            'job_types': dict,
-            'date': dict,
-            'positions': list,
-            'locations': list,
-            'location_blacklist': list,
-            'distance': int,
-            'company_blacklist': list,
-            'title_blacklist': list,
+            "remote": bool,
+            "experience_level": dict,
+            "job_types": dict,
+            "date": dict,
+            "positions": list,
+            "locations": list,
+            "location_blacklist": list,
+            "distance": int,
+            "company_blacklist": list,
+            "title_blacklist": list,
+            "llm_model_type": str,
```

Comment:
line 84, 85 -> they are removed - check base branch

Reply:
revert this change

Reply:
done